plugins/Kaleidoscope-Colormap-Overlay/src/kaleidoscope/plugin/Colormap-Overlay.cpp

```diff
@@ -0,0 +1,79 @@
+/* Kaleidoscope-Colormap-Overlay - A LED plugin for Kaleidoscope.
+ * Copyright (C) 2017-2018  Keyboard.io, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "kaleidoscope/plugin/Colormap-Overlay.h"
+
+#include <stdint.h>  // for uint8_t
+
+#include "kaleidoscope/KeyAddr.h"               // for KeyAddr, MatrixAddr, MatrixAddr<>::...
+#include "kaleidoscope/device/device.h"         // for cRGB, CRGB
+#include "kaleidoscope/event_handler_result.h"  // for EventHandlerResult, EventHandlerRes...
+#include "kaleidoscope/key_defs.h"              // for Key, KEY_FLAGS, Key_NoKey, LockLayer
+#include "kaleidoscope/layers.h"                // for Layer, Layer_
+#include "kaleidoscope/plugin/LEDControl.h"     // for LEDControl
+#include <Kaleidoscope-LED-Palette-Theme.h>     // for LEDPaletteTheme
+
+namespace kaleidoscope {
+namespace plugin {
+uint16_t ColormapOverlay::map_base_;
+
+void ColormapOverlay::setup() {
+  map_base_ = ::LEDPaletteTheme.reserveThemes(1);
+}
```

Comment:
As stated above, I would prefer to remove this and move that single line to [`ColormapOverlay::onSetup()`](https://github.com/keyboardio/Kaleidoscope/pull/1359/files#diff-b0e992db6b7067570f61bfaaf670c1995074526312e862f9fca49be6f04053bbR53), but that doesn't work as I expect it to.

I'd appreciate some help